2312/2/battle.rb

```diff
@@ -0,0 +1,67 @@
+require './line'
+require 'russian_obscenity'
+
+# this class analyzes text of battle
+class Battle
+  attr_reader :battle_file_path
+
+  def initialize(battle_file_path)
+    @battle_file_path = battle_file_path
+  end
+
+  def paired?
+    @battle_file_path.partition(/( против | vs )/).first.include?(' & ')
+  end
+
+  def process_battle(info, rapper)
+    info[1] += count_rounds
+    info[6] += all_words(rapper).size
+    info[3] += bad_words_amount(rapper)
+    info
+  end
+
+  private
+
+  def count_rounds
+    rounds = rounds_amount
+    rounds.zero? ? 1 : rounds
+  end
+
+  def rounds_amount(counter = 0)
+    IO.read(@battle_file_path).each_line { |line| counter += 1 if line[/Раунд \d/] }
+    counter
+  end
+
+  def bad_words_amount(rapper)
+    counter = 0
+    all_words(rapper).each { |word| counter += 1 if word.include?('*') || RussianObscenity.obscene?(word) }
```

Comment:
```ruby
all_words_said_by(rapper_name).select { |word| word.include?('*') || RussianObscenity.obscene? }.count
```
Even more better to introduce `Word` class which will respond to the `#obscene?` message.
Then the code will look like this:
```ruby
all_words_said_by(rapper_name) # returns an array of Word class objects

all_words_said_by(rapper_name).select(&:obscene?).count # returns obscene words
```

Do not forget about useful Enumerable methods!